packages/components/src/components/arrow-indicator/arrow-indicator.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Icon from '../icon/icon';
+
+type TArrowIndicatorProps = {
+    classname?: string;
+    value?: number | string;
+};
+
+type TData = {
+    previous_icon: string;
+    icon: string;
+    previous_value?: string | number;
+    value?: string | number;
+};
+
+const ArrowIndicator = ({ classname, value }: TArrowIndicatorProps) => {
+    const [is_visible, setIsVisible] = React.useState(false);
+    const [data, setData] = React.useState<TData>({
+        icon: '',
+        previous_icon: '',
+    });
+    const { icon, previous_icon, previous_value } = data;
+    const has_comparable_values = !isNaN(Number(data.value)) && !isNaN(Number(value));
+    const timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    React.useEffect(() => {
+        setIsVisible(true);
+
+        if (data.value !== Number(value)) {
+            setData(prev_data => {
+                const has_increased = Number(prev_data.value) < Number(value);
+                const icon_name = has_increased ? 'IcProfit' : 'IcLoss';
+                return {
+                    icon: has_comparable_values ? icon_name : '',
+                    previous_icon: prev_data.icon,
+                    previous_value: prev_data.value,
+                    value,
+                };
+            });
+
+            clearTimeout(timeout_id.current);
+            timeout_id.current = setTimeout(() => {
+                setIsVisible(false);
+            }, 3000);
```

Comment:
whats the criteria in consideration for setting to 3 seconds

Reply:
@henry-deriv this requirement was approved by PO and listed in my task requirements :) `ArrowIndicator` should hide if the value is unchanged before 3 seconds pass.

Reply:
i thought it was for switching every 3 seconds